src/Model/Map.php

```diff
@@ -2,15 +2,216 @@
+				$thumbnail = wp_get_attachment_image_url( $thumbnailID, 'thumbnail' );
+				$images    = [
+					'thumbnail' => wp_get_attachment_image_src( $thumbnailID, 'thumbnail' ),
+					'medium'    => wp_get_attachment_image_src( $thumbnailID, 'medium' ),
+					'large'     => wp_get_attachment_image_src( $thumbnailID, 'large' ),
+					'full'      => wp_get_attachment_image_src( $thumbnailID, 'full' ),
+				];
+				$items[]   = [
+					'id'         => $item->ID,
+					'name'       => $item->post_title,
+					'short_desc' => has_excerpt( $item->ID ) ? wp_strip_all_tags( get_the_excerpt( $item->ID ) ) : "",
+					'status'     => $item->post_status,
+					'terms'      => $item_terms,
+					'link'       => add_query_arg( 'cb-location', $post->ID, get_permalink( $item->ID ) ),
+					'thumbnail'  => $thumbnail ?: null,
+					'images'     => $images,
+					'timeframes' => $timeframesData
+				];
+			}
+
+			if ( count( $items ) ) {
+				$locations[ $post->ID ] = [
+					'lat'           => (float) $location_meta['geo_latitude'][0],
+					'lon'           => (float) $location_meta['geo_longitude'][0],
+					'location_name' => $post->post_title,
+					'location_link' => get_permalink( $post->ID ),
+					'closed_days'   => unserialize( $closed_days ),
+					'address'       => [
+						'street' => $location_meta[ COMMONSBOOKING_METABOX_PREFIX . 'location_street' ][0],
+						'city'   => $location_meta[ COMMONSBOOKING_METABOX_PREFIX . 'location_city' ][0],
+						'zip'    => $location_meta[ COMMONSBOOKING_METABOX_PREFIX . 'location_postcode' ][0],
+					],
+					'items'         => $items,
+				];
+
+				if ( $show_location_contact ) {
+					$locations[ $post->ID ]['contact'] = $location_meta[ COMMONSBOOKING_METABOX_PREFIX . 'location_contact' ][0];
+				}
+			}
+		}
+
+		return $locations;
+	}
+
+	/**
+	 * recursive clean up of location data entries
+	 *
+	 * @param $value
+	 * @param $linebreak_replacement
+	 *
+	 * @return mixed|string|string[]|null
+	 */
+	public static function cleanup_location_data_entry( $value, $linebreak_replacement ) {
+
+		if ( is_string( $value ) ) {
+			$value = wp_strip_all_tags( $value ); //strip all tags
+			$value = preg_replace( '/(\r\n)|\n|\r/', $linebreak_replacement, $value ); //replace linebreaks
+		}
+
+		if ( is_array( $value ) ) {
+			foreach ( $value as &$child_value ) {
+				//recursive call
+				$child_value = self::cleanup_location_data_entry( $child_value, $linebreak_replacement );
+			}
+		}
+
+		return $value;
+	}
+
+	/**
+	 * clean up the location data
+	 *
+	 * @param $locations
+	 * @param $linebreak_replacement
+	 *
+	 * @return mixed
+	 */
+	public static function cleanup_location_data( $locations, $linebreak_replacement ) {
+		foreach ( $locations as &$location ) {
+			$location = self::cleanup_location_data_entry( $location, $linebreak_replacement );
+		}
+
+		return $locations;
+	}
+
+	/**
+	 * basic check if the given string is valid JSON
+	 **/
+	public static function is_json( $string ): bool {
```

Comment:
Mh wieso ist das public? Als ob es wiederverwendbar im ganzen Projekt sein muss.

Reply:
ich finde die Aufrufe dazu nicht.

Reply:
Das wird auch in der MapData::geo_search() referenziert.

Reply:
Wenn wir Code so (als "public static") veröffentlichen müssen wir diesen auch in Zukunft warten. Zusätzlich finde ich den Code jetzt nicht unbedingt so komplex, dass er eine eigene Methode bekommen sollte. Auch da kaum an anderer Stelle verwendet. Da die Methode nur in MapData aufgerufen wirden, wäre ich dafür ihn entweder linline direkt in MapData zu verwenden oder eine private Methode in MapData zu definieren. Falls du dich für letzteres entscheidest, kannst du den Unit-Test trotzdem lassen.